save-frontend/src/main/kotlin/com/saveourtool/save/frontend/utils/Action.kt

```diff
@@ -135,18 +141,35 @@ external interface ButtonWithActionProps : Props {
 
     /**
      * Modal buttons
+     *
+     * @param action is the main action of the buttons
+     * @param closeWindow is the action of closing the window and assigning the status false to the checkBox
+     * @param ChildrenBuilder
+     * @param isClickMode is checkBox status
+     * @param setButtonNumber with a large number of buttons, it allows to identify the button using a number
+     * @return buttons
      */
     @Suppress("TYPE_ALIAS")
-    var modalButtons: (action: () -> Unit, WindowOpenness, ChildrenBuilder) -> Unit
+    var modalButtons: (
+        action: () -> Unit,
+        closeWindow: () -> Unit,
+        ChildrenBuilder,
+        isClickMode: Boolean,
+        setButtonNumber: StateSetter<Int>,
+    ) -> Unit
 
     /**
      * Condition for click
      */
     var conditionClick: Boolean
 
     /**
-     * Request
+     * function passes arguments to call the request
+     *
+     * @param isClickMode is checkBox status
+     * @param buttonNumber is number of the button pressed
+     * @return lazy response
      */
     @Suppress("TYPE_ALIAS")
-    var sendRequest: (Boolean) -> DeferredRequestAction<Response>
+    var sendRequest: (isClickMode: Boolean, buttonNumber: Int) -> DeferredRequestAction<Response>
```

Comment:
what's buttonNumber? can we send the object itself which we want to delete?

Reply:
I'm trying to make the logic more universal. In it, you can not only delete. 
I do not know how else to convey which specific button we pressed in order to further describe the specific logic of this button, namely a request for a backend and some kind of change in the frontend states, in case of a successful request.

Reply:
Right now I just don't have any other ideas how to identify the buttons. You can delete these identifiers and leave the idea with 2 buttons, it seems to me that it is most often used, but I wanted to make a universal one.

Reply:
I don't see where buttonNumber is used and where it's set.
I see `sendRequest` and `onActionSuccess` which take `buttonNumber` as argument, but it's used as `_` in all places where the prop variable is written.
I see `modalButtons` which takes `setButtonNumber` as argument, but it's used as `_` in all places where the prop variable is written.

So it's not clear to me what do you try to resolve using this number

Reply:
Done

Reply:
What have you done?